vllm/v1/core/sched/scheduler.py

```diff
@@ -616,25 +614,24 @@ def update_from_output(
 
             stopped = False
             new_logprobs = None
-            new_token_ids: list[int] = []
+            new_token_ids = generated_token_ids
 
             # Append generated tokens and check for stop. Note that if
             # a request is still being prefilled, we expect the model runner
             # to return empty token ids for the request.
-            for output_token_id in generated_token_ids:
+            for num_new, output_token_id in enumerate(new_token_ids, 1):
                 request.append_output_token_ids(output_token_id)
-                new_token_ids.append(output_token_id)
 
                 # Check for stop and update request state.
                 # This must be called before we make the EngineCoreOutput.
                 stopped = check_stop(request, self.max_model_len)
                 if stopped:
                     self._free_request(request)
+                    del new_token_ids[num_new:]  # Trim new tokens if needed.
```

Comment:
Honestly I feel this is a bit over-optimization that hurts readability. Does it make a noticeable performance difference?

Reply:
Unlikely to make a noticeable difference by itself, just adds incrementally to the work... 99.9% of times there's no need to copy the list.

To me at least it's quite readable since we are just trimming any remaining tokens from the list in the case that we stop early. But happy to revert it!

Reply:
Thanks for the clarification. I'm ok with this. :)

Reply:
Another way to look at it, we save allocating 1000 new lists per step for large batch sizes